app/src/main/java/com/lastaoutdoor/lasta/viewmodel/WeatherViewModel.kt

```diff
@@ -63,11 +68,33 @@ constructor(
 
       // Call surrounded by try-catch block to make handle exceptions caused by Weather API
       try {
+        // current weather for activity location
         val weather = weatherRepository.getWeatherWithLoc(a.startPosition.lat, a.startPosition.lon)
         _weather.postValue(weather)
+        // forecast for activity location

```

Comment:
Great reflex to put it inside the try block, so the ErrorToast handle the error :)